bundle/Command/MigrateCommand.php

```diff
@@ -0,0 +1,429 @@
+            }
+            foreach ($campaignData->mailings as $mailingData) {
+                $mailing = new Mailing();
+                $mailing->setNames((array) $mailingData->names);
+                $mailing->setStatus($mailingData->status);
+                $mailing->setRecurring(false);
+                $mailing->setHoursOfDay([$mailingData->hoursOfDay]);
+                $mailing->setDaysOfMonth([$mailingData->daysOfMonth]);
+                $mailing->setMonthsOfYear([$mailingData->monthsOfYear]);
+                $mailing->setLocationId($mailingData->locationId);
+                $mailing->setSiteAccess($mailingData->siteAccess);
+                $mailing->setSubject($mailingData->subject);
+                $this->entityManager->persist($mailing);
+                $campaign->addMailing($mailing);
+                ++$mailingCounter;
+            }
+            $this->entityManager->persist($campaign);
+        }
+
+        // Importing Users & Subscriptions
+        foreach ($fileNames->users as $userFile) {
+            $userData = json_decode($this->ioService->readFile('ezmailing/user/'.$userFile.'.json'));
+
+            // check if email already exists
+            $existingUser = $userRepository->findOneBy(['email' => $userData->email]);
+            if (null === $existingUser) {
+                $user = new User();
+                $user
+                    ->setEmail($userData->email)
+                    ->setBirthDate($userData->birthDate)
+                    ->setCompany($userData->company)
+                    ->setFirstName($userData->firstName)
+                    ->setLastName($userData->lastName)
+                    ->setStatus($userData->status)
+                    ->setOrigin('site');
+
+                foreach ($userData->subscriptions as $subscription) {
+                    if (\array_key_exists($subscription->list_contentobject_id, $listIds)) {
+                        $registration = new Registration();
+                        /* @var MailingList $mailingList */
+                        $mailingList = $mailingListRepository->findOneBy(
+                            ['id' => $listIds[$subscription->list_contentobject_id]]
+                        );
+                        $registration->setMailingList($mailingList);
+                        $registration->setApproved($subscription->approved);
+                        $user->addRegistration($registration);
+                        ++$subscriptionCounter;
+                    }
+                }
+                $this->entityManager->persist($user);
+                ++$userCounter;
+            }
+        }
+        $this->entityManager->flush();
+
+        $this->io->section(
+            'Total: '.$listCounter.' lists, '.$mailingCounter.' mailings, '.$userCounter.' users, '.
+            $subscriptionCounter.' registrations.'
+        );
+        $this->io->success('Import done.');
+    }
+
+    private function clean(): void
```

Comment:
I think what you are doing `TRUNCATE` please check and change ;)

Reply:
Don't use TRUNCATE because of FOREIGN KYES constraints. Added the comment about that